### PR TITLE
docs enhancement for binv, cmpbqm, sinv, dxinit, moda_bufrsr

### PR DESCRIPTION
**Pull request by jack-woollen**
a few docs upgrades to test the waters.

### CODE REVIEW

**Comment by jbathegit** [comment]
Thanks Jack and this looks good, but did you happen to see Ed's separate PR #278 where he figured out a fix to issue #277?  He was able to make some minor tweaks to the sinv and cmpbqm codes to get rid of some doxygen warnings, so it would be good to get those in as well.  And I'm guessing you both started from the same baseline version of those codes when making your changes, which means we're on track for a merge conflict one way or the other in the near future.

So is there any way you could incorporate your changes with Ed's into a single consolidated version of the sinv.f90.in and cmpbqm.f90 files?  If it helps, he listed the exact commands he used to make his fixes within the conversation text of #277.

Please let me know - thanks!

File: src/moda_bufrsr.F [hunk 1]
```diff
@@ -35,7 +35,7 @@ MODULE MODA_BUFRSR
 	  INTEGER :: JNOD
 C>        Section 1 date-time of BUFR message.
 	  INTEGER :: JDAT
-C>        ???
+C>        Indicator of stack status when entering subroutine rewnbf.
 	  INTEGER, ALLOCATABLE :: JSR(:)
 C>        BUFR message.
```
Comment (jbathegit) [inline src/moda_bufrsr.F:40]: Could you please change the commented reference to the subroutine name from `rewnbf` to `rewnbf()`?  That way, doxygen will generate an automatic link to the documentation for that subroutine from this module.

Comment (jack-woollen) [reply src/moda_bufrsr.F:40]: done

Comment (jbathegit) [reply src/moda_bufrsr.F:40]: Thanks Jack!

**Comment by jbathegit** [comment]
Hey Jack, just to clarify, is it still your plan to apply the same fixes Ed made to his versions of sinv.f90.in and cmpbqm.f90 on top of your versions of those same source files within this PR?  That was my understanding from your comments in #278, but maybe I misunderstood.

**Comment by jack-woollen** [comment]
Yup, that's what I'm working on. Almost done.

**Comment by jbathegit** [comment]
OK got it.  Then in that case, and once this is finished and merged, then #278 will become obsolete and can just be withdrawn.

**Comment by jbathegit** [comment]
Hey Jack,  as I noted earlier, I'm out of steam for today, and my wife and daughter are now bugging me about dinner ;-)  So I'm going to cut out and can get back to this on Monday, and I'll review your final commits for this PR at that time.   Have a good weekend!

**Comment by jack-woollen** [comment]
That's fine. Y'all have a good weekend too!

**Comment by jack-woollen** [comment]
@jbathegit @edwardhartnett I added some more files to this branch but now I can't push anymore because of some conflict. Maybe if you approve the PR and merge the changes it will straighten itself out.  What do you all think?

**Comment by jbathegit** [comment]
Hi @jack-woollen, just within the past hour I downloaded and viewed the resulting HTML (doxygen) output that was generated from this PR, and after seeing that I made a few tweaks in a follow-on commit, so that's probably why you're now seeing a merge conflict.

I was just about to approve and merge your PR with my additional tweaks.   How about I go ahead and do that, and then after that you can do a fetch back into your local branch and then reconcile and push up any further changes you want to make? 

Sorry if I messed you up, but there were a few details that still needed to be cleaned up, and I wasn't aware that you were still working on this PR or else I would have just used a bunch of comments to report back to you what I'd found and let you do the edits yourself.

**Comment by jack-woollen** [comment]
@jbathegit Sounds good. I'll try again when I'm back from some errands.

**Review by jbathegit — APPROVED** [review]

**Comment by jbathegit** [comment]
OK, the changes have been merged, but the branch hasn't been deleted if you want to go ahead and add anything else.  Just let me know and I'll take a look when I get a chance.

In the meantime I'll keep working on other PRs that have been submitted ;-)

**Pull request merged**